packages/components/src/range-control/utils.ts

```diff
@@ -46,7 +54,7 @@ export function useControlledRangeValue( {
 	max,
 	value: valueProp,
 	initial,
-} ) {
+}: useControlledRangeValueProps ): useControlledRangeValueReturn {
```

Comment:
This is a nit, so feel free to ignore — but in general, I prefer to use as least hardcoded types as possible, and let the TypeScript engine infer the rest.

In this case, we could avoid the return type and rewrite the code like this

<details>

<summary>Click to expand</summary>

```diff
diff --git a/packages/components/src/range-control/types.ts b/packages/components/src/range-control/types.ts
index 0fa4c45118..eda285376a 100644
--- a/packages/components/src/range-control/types.ts
+++ b/packages/components/src/range-control/types.ts
@@ -270,18 +270,13 @@ export type TrackProps = {
 	trackColor: CSSProperties[ 'color' ];
 };
 
-export type useControlledRangeValueProps = {
+export type useControlledRangeValueArgs = {
 	initial?: number;
 	max: number;
 	min: number;
 	value: number | null;
 };
 
-export type useControlledRangeValueReturn = [
-	number | '' | null,
-	( nextValue: number | null ) => void
-];
-
 export type useMarksArgs = NumericProps & {
 	marks: RangeMarks;
 	step: number;
diff --git a/packages/components/src/range-control/utils.ts b/packages/components/src/range-control/utils.ts
index 7a455134a1..2e55db6f7d 100644
--- a/packages/components/src/range-control/utils.ts
+++ b/packages/components/src/range-control/utils.ts
@@ -14,10 +14,7 @@ import { useCallback, useRef, useEffect, useState } from '@wordpress/element';
 import { useControlledState } from '../utils/hooks';
 import { clamp } from '../utils/math';
 
-import type {
-	useControlledRangeValueProps,
-	useControlledRangeValueReturn,
-} from './types';
+import type { useControlledRangeValueArgs } from './types';
 
 const noop = () => {};
 
@@ -54,14 +51,14 @@ export function useControlledRangeValue( {
 	max,
 	value: valueProp,
 	initial,
-}: useControlledRangeValueProps ): useControlledRangeValueReturn {
+}: useControlledRangeValueArgs ) {
 	const [ state, setInternalState ] = useControlledState(
 		floatClamp( valueProp, min, max ),
 		{ initial, fallback: null }
 	);
 
 	const setState = useCallback(
-		( nextValue ) => {
+		( nextValue: number | null ) => {
 			if ( nextValue === null ) {
 				setInternalState( null );
 			} else {
@@ -71,7 +68,7 @@ export function useControlledRangeValue( {
 		[ min, max ]
 	);
 
-	return [ state, setState ];
+	return [ state, setState ] as const;
 }
 
 /**
```
</details>